cwltool/job.py

```diff
@@ -175,13 +176,24 @@ def _setup(self, runtimeContext: RuntimeContext) -> None:
         if not os.path.exists(self.outdir):
             os.makedirs(self.outdir)
 
+        def is_streamable(file: str) -> bool:
+            if not runtimeContext.streaming_allowed:
+                return False
+            for inp in self.joborder.values():
+                if isinstance(inp, dict) and inp.get("location", None) == file:
+                    return inp.get("streamable", False)
+            return False
```

Comment:
This line needs testing

Reply:
I think this is dead code. Maybe I should add an "It should not get here" comment? I no not know an use-case when the file is not in self.joborder and that line would be reached. Another way of writing this function is to remove the first 2 lines and instead of `False` return `runtimeContext.streaming_allowed`. But doing this would unnecessarily iterate through all the files even when streaming is not allowed.

Reply:
A `File` from an input `Directory` could be staged in using `InitialWorkDirRequirement` and it itself would not appear in the top level of the joborder, and so would (incorrectly) reach this line. Maybe throw an exception here and run all the conformance tests and regular on a version of the code with `runtimeContext.streaming_allowed` set to `True` to see if there are other scenarios you are missing.

Reply:
If it returns False an exception is thrown here:
https://github.com/common-workflow-language/cwltool/blob/5c4464ea4670e30fcd1f22e46b6dc98ab17d2fe1/cwltool/job.py#L193

Reply:
Okay, then temporarily set `runtimeContext.streaming_allowed` to always be `True` and run the pytest and conformance tests again